File: dataset/javascript/original/ssrf.js
const express = require('express');
const axios = require('axios');
const app = express();

app.get('/fetch', async (req, res) => {
  const url = req.query.url;
  try {
    const response = await axios.get(url);
    res.json({
      status: response.status,
      body: response.data.toString().substring(0, 1000)
    });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

File: dataset/python/original/ssrf.py
import requests
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route("/fetch")
def fetch_url():
    url = request.args.get("url")
    response = requests.get(url)
    return jsonify({
        "status": response.status_code,
        "body": response.text[:1000]
    })
